analyses/fusion-frequencies/01-fusion-frequencies.R

```diff
@@ -0,0 +1,242 @@
+primary_indp_sdf <- read_tsv(primary_independence_list,
+  col_types = cols(
+    .default = col_guess()))
+
+# relapse independent samples
+relapse_indp_sdf <- read_tsv(relapse_independence_list,
+  col_types = cols(
+    .default = col_guess()))
+
+# Subset independent samples in histology table --------------------------------
+message('Prepare data...')
+# td = tumor descriptor
+td_htl_dfs <- list(
+  primary_htl_df = htl_df %>%
+    filter(Kids_First_Biospecimen_ID %in%
+             primary_indp_sdf$Kids_First_Biospecimen_ID),
+  
+  relapse_htl_df = htl_df %>%
+    filter(Kids_First_Biospecimen_ID %in%
+             relapse_indp_sdf$Kids_First_Biospecimen_ID),
+  
+  overall_htl_df = htl_df %>%
+    filter(sample_type == 'Tumor')
+)
+
+# keep only samples with non-NA cancer_group and cohort
+# add cancer_group_cohort column
+td_htl_dfs <- lapply(td_htl_dfs, function(x) {
+  # assert all Kids_First_Biospecimen_IDs are unique
+  stopifnot(identical(nrow(x), length(unique(x$Kids_First_Biospecimen_ID))))
+  
+  fx <- filter(x, !is.na(cancer_group), !is.na(cohort))
+  return(fx)
+})
+
+# Subset tumor samples to filter alteration df ---------------------------
+tumor_kfbids <- htl_df %>%
+  filter(sample_type == 'Tumor') %>%
+  pull(Kids_First_Biospecimen_ID)
+
+
+fusion_df <- fusion_df  %>%
+  # We want to keep track of the gene symbols for each sample-fusion pair
+  dplyr::select(Sample,
+                Kids_First_Participant_ID, 
+                FusionName, 
+                Gene1A, 
+                Gene1B, 
+                Gene2A, 
+                Gene2B, 
+                Fusion_Type,
+                DomainRetainedGene1A,
+                DomainRetainedGene1B,
+                reciprocal_exists,
+                annots,
+                BreakpointLocation,
+                ends_with("anno")) %>%
+  # Update reciprocal_exits to fusion with atleast 1 kinase gene involved
+  mutate(
+         reciprocal_exists_kinase = 
+           # if either gene are kinase we will have "Yes" or "No" values
+           # in DomainRetainedGene1A and DomainRetainedGene1B
+           if_else(is.na(DomainRetainedGene1A) &
+                     is.na(DomainRetainedGene1B) ,
+                   # reciprocl_exists is a logical column
+                   FALSE,
+                   reciprocal_exists)) %>%
+  dplyr::rename("Kids_First_Biospecimen_ID"="Sample",
+         "Kinase_domain_retained_Gene1A" = "DomainRetainedGene1A",
+         "Kinase_domain_retained_Gene1B" = "DomainRetainedGene1B",
+         "Reciprocal_exists_either_gene_kinase" = "reciprocal_exists_kinase")%>%
+  # replace NA to "" in columns that have NA
+  replace_na(list("Kinase_domain_retained_Gene1A"="",
+                  "Kinase_domain_retained_Gene1B"="",
+                  "Reciprocal_exists_either_gene_kinase"="",
+                  "Gene1A_anno"="",
+                  "Gene1B_anno"="",
+                  "Gene2A_anno"="",
+                  "Gene2B_anno"="",
```

Comment:
I am still seeing NAs in these two columns when I read this in:
```
fus <- read_tsv("~/Documents/GitHub/ped-ot/OpenPedCan-analysis/analyses/fusion-frequencies/results/putative-oncogene-fusion-freq.tsv", 
                na=character(0))
```
but the other columns are blanks, as expected. Could be because Gene2A_anno and Gene2B_anno are of class `logical` instead of `character`. Perhaps, you may need to apply @logstar's solution as in [this commit 38629de](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/45/commits/38629de5fa8fd6acb863056e9a46d8262558d790) for these columns.

Reply:
Thanks for the review! 
I looked into the NAs in Gene2A_anno and Gene2B_anno, those should not be all NA. I also see it in `analyses/fusion_filtering/results/fusion-putative-oncogenic.tsv` so an upstream fix in fusion_filtering will be needed here. 

The code from @logstar is already part of the code ( basically all code here is adapted from the code from #45 )

Reply:
I don't think we need to have this in the `fusion-putative-oncogenic.tsv` but only in the FNL-specific files - I prefer having NAs in files rather than blanks in general, but this is their specific request, so I think it should only apply to their files.

Reply:
Right, I get that the NA to blank is specific for FNL files. 

I wanted to just mention that the original `fusion-putative-oncogenic.tsv` had Gene2A_anno and Gene2B_anno as all `NA` which is not expected  because oncogene like MET are Gene2A/B. So I went back and added `guess_max` to read the files correctly in #50 and now `fusion-putative-oncogenic.tsv` has non-NA values in `*_anno` columns. 

After I updated the #50 I also reran the code here and now seems to have spaces in `Gene2A_anno` and `Gene2B_anno` when I read it as `fus <- read_tsv("~/Documents/PedOpenTargets/OpenPBTA-analysis/analyses/fusion-frequencies/results/putative-oncogene-fusion-freq.tsv",na=character(0))`

```
> fus$Gene2A_anno %>% unique()
[1] ""                                                  "TranscriptionFactor"                              
[3] "Oncogene, Kinase"                                  "Oncogene"                                         
[5] "TumorSuppressorGene"                               "Kinase, TumorSuppressorGene, CosmicCensus, Kinase"
[7] "TumorSuppressorGene, Oncogene"                     "Kinase"                                           
[9] "Kinase, CosmicCensus, Oncogene, Kinase"         
```

Let me know if I missed anything.

Reply:
OH, I see I see - gotcha. I misunderstood!